src/neo/SmartContract/InteropService.Native.cs

```diff
@@ -7,12 +7,12 @@ partial class InteropService
     {
         internal static class Native
         {
-            public static readonly InteropDescriptor Deploy = Register("Neo.Native.Deploy", Native_Deploy, 0, TriggerType.Application);
+            public static readonly InteropDescriptor Deploy = Register("Neo.Native.Deploy", Native_Deploy, 0, TriggerType.Application, CallFlags.AllowModifyStates);
 
             static Native()
             {
                 foreach (NativeContract contract in NativeContract.Contracts)
-                    Register(contract.ServiceName, contract.Invoke, contract.GetPrice, TriggerType.System | TriggerType.Application);
+                    Register(contract.ServiceName, contract.Invoke, contract.GetPrice, TriggerType.System | TriggerType.Application, CallFlags.AllowCall | CallFlags.AllowModifyStates);
```

Comment:
Some native contract methods do not modify state, such as `balanceOf`. The flags are checked here:

https://github.com/neo-project/neo/blob/158af1b892562dc9a8c4e4edbc4bb73aae739b52/src/neo/SmartContract/Native/NativeContract.cs#L96-L98

```suggestion
                    Register(contract.ServiceName, contract.Invoke, contract.GetPrice, TriggerType.System | TriggerType.Application, CallFlags.AllowCall);
```